### ISSUE TITLE
Using search box causes Segmentation fault.

### ISSUE THREAD

**Issue by ZuluEchoRomeOscar** [VCVRack/Rack#652]
OS (if applicable): MX Linux 17 (Debian 9.3 base)
Version (or "dev" if compiling from source): 0.5.1

After updating all the plugins in .Rack via Chris Choltes' 'vcv-plugindownloader' script [ https://github.com/cschol/vcv-plugindownloader ], I'm left with a Rack that crashes whenever I attempt to type into the plugin search box. Launching via the terminal returns this fault:

./Rack.sh: line 2: 25329 Segmentation fault      LD_LIBRARY_PATH=. ./Rack

It seems unlikely that the script caused this fault - but possibly one of the updated plugins? Several were updated at the time, so it would be difficult for me to determine which, but the script only downloads updated plugins from the community repository which have a Linux binary available, and this behaviour only started after my first use of the script - about midnight (GMT) 28/1/18.

**Comment by AndrewBelt** [comment]
Can you narrow it down and confirm that the issue is solved when there are no plugins in `plugins/`? Do a binary search on the plugins to see which one causes it.

**Comment by ZuluEchoRomeOscar** [comment]
I'm running 0.5.1 - not a self-build, so no `make run debug` I'm afraid.
Can confirm that the issue is solved when there are no plugins.
I'm afraid I'm not familiar with the task of 'making a binary search' on the plugins....

**Comment by AndrewBelt** [comment]
Binary search = Remove half of them, test, if failure, remove half again. If success, put the other half back, test, etc.

**Comment by ZuluEchoRomeOscar** [comment]
Right. Actually Intuited what you were meaning, and started on removing one-by-one. Hit pay dirt on the first one - AepelzensModules are the guilty party!

**Comment by ZuluEchoRomeOscar** [comment]
Which is peculiar, since there are not binaries of those modules for the script to pull down - which means I acquired it elsewhere. Probably on the Facebook group. Which means I'd had it a while before the bug appeared.

**Comment by ZuluEchoRomeOscar** [comment]
Belay that - the issue has mysteriously reappeared, despite AepelzensModules no longer being in the plugins directory. I can see I'm going to have to do this more methodically. But later - when my brain is awake.

**Comment by cschol** [comment]
It is highly likely that this is related to this [issue](https://github.com/av500/vcvrackplugins_av500/issues/12) that I logged against the `av500` plugin (compiled with `-fshort-enums`).

Remove the `-fshort-enums` from the `Makefile` and rerun `make` inside the `av500.git` directory (I assume you are running `vcv-plugindownloader` with the `--source` option to compile from source.

[Note: I wrote the `vcv-plugindownloader` and this is not related to the script.]

**Comment by ZuluEchoRomeOscar** [comment]
Actually, I've been unable to reproduce the fault in the last day of testing - whatever combination of plugins I leave in the directory. Something specific (and transitory?) to my system perhaps.

**Comment by AndrewBelt** [comment]
Yeah, it's definitely `-fshort-enums`.

**Comment by ZuluEchoRomeOscar** [comment]
Okay - I've removed av500 from my folder. Will report back if it crashes the same way again.

**Comment by cschol** [comment]
@ZuluEchoRomeOscar Did you have any more issues with this after removing the `av500` plugin?
